tests/tizen_capi/unittest_tizen_capi.cc

```diff
@@ -10000,6 +10000,208 @@ TEST (nnstreamer_capi_custom, register_filter_10_n)
+    uintarray[i][2] = i + 3;
+    uintarray[i][3] = i + 2;
+  }
+
+  status = ml_pipeline_src_get_tensors_info (srchandle, &info);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  ml_tensors_info_get_count (info, &count);
+  EXPECT_EQ (count, 1U);
+
+  ml_tensors_info_get_tensor_type (info, 0, &type);
+  EXPECT_EQ (type, ML_TENSOR_TYPE_UINT8);
+
+  status = ml_tensors_data_create (info, &data);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  /* Set tensor data and push buffers to source pad */
+  for (i = 0; i < 10; i++) {
+    status = ml_tensors_data_set_tensor_data (data, 0, uintarray[i], 4);
+    EXPECT_EQ (status, ML_ERROR_NONE);
+
+    status = ml_pipeline_src_input_data (srchandle, data, ML_PIPELINE_BUF_POLICY_DO_NOT_FREE);
+    EXPECT_EQ (status, ML_ERROR_NONE);
+
+    g_usleep (50000); /* 50ms. Wait a bit. */
+  }
+
+  status = ml_pipeline_stop (pipe);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  status = ml_pipeline_src_release_handle (srchandle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  status = ml_pipeline_sink_unregister (sink_false);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  status = ml_pipeline_destroy (pipe);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  status = ml_pipeline_tensor_if_custom_unregister (custom);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  EXPECT_TRUE (g_file_get_contents (file, (gchar **)&content, &len, NULL));
+  EXPECT_EQ (len, 4U * 5);
+  EXPECT_TRUE (content != nullptr);
+
+  /* Check if the TRUE path data is received correctly.  */
+  if (content && len == 20) {
+    for (i = 0; i < 5; i++) {
+      EXPECT_EQ (content[i * 4 + 0], i + 4);
+      EXPECT_EQ (content[i * 4 + 1], i + 1);
+      EXPECT_EQ (content[i * 4 + 2], i + 3);
+      EXPECT_EQ (content[i * 4 + 3], i + 2);
+    }
+  }
+  g_free (content);
+
+  /* The FALSE path receives 5 buffers. */
+  EXPECT_EQ (*count_sink, 5U);
+
+  for (i = 0; i < 10; i++) {
+    g_free (uintarray[i]);
+  }
+  ml_tensors_info_destroy (info);
+  ml_tensors_data_destroy (data);
+  g_free (pipeline);
+  g_free (count_sink);
+}
+
+/**
+ * @brief Test for tensor_if custom registration.
+ * @detail Invalid params.
+ */
+TEST (nnstreamer_capi_if, register_01_n)
+{
+  ml_pipeline_if_h custom;
+  int status;
+
+  /* test code with null param */
+  status = ml_pipeline_tensor_if_custom_register (NULL, test_if_custom_cb, NULL, &custom);
+  EXPECT_NE (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Test for tensor_if custom registration.
+ * @detail Invalid params.
+ */
+TEST (nnstreamer_capi_if, register_02_n)
+{
+  ml_pipeline_if_h custom;
+  int status;
+
+  /* test code with null param */
+  status = ml_pipeline_tensor_if_custom_register ("tifx", NULL, NULL, &custom);
+  EXPECT_NE (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Test for tensor_if custom registration.
+ * @detail Invalid params.
+ */
+TEST (nnstreamer_capi_if, register_03_n)
+{
+  int status;
+
+  /* test code with null param */
+  status = ml_pipeline_tensor_if_custom_register ("tifx", test_if_custom_cb, NULL, NULL);
```

Comment:
lets cover another cases in this function (name or cb is null)

Reply:
resolved.